src/node_hybrid.cpp

```diff
@@ -471,6 +486,26 @@ inline float distanceHeuristic2D(
   return std::sqrt(dx * dx + dy * dy);
 }
 
+void NodeHybrid::initializeFootprintCollisionMap(const std::shared_ptr<costmap_2d::Costmap2DROS>& costmap_ros)
+{
+    const costmap_2d::Costmap2D* costmap = costmap_ros->getCostmap();
```

Comment:
Q/R: You are copying the whole global costmap here again? if yes, this might be very expensive in some cases (e.g. in large maps).
Isn't this the same problem we had before and you handle it by creating a new map with the desired size (the search space bounds), instead of using all the map?